tests/torch/test_algo_common.py

```diff
@@ -490,6 +491,8 @@ def test_target_device_is_propagated_to_algos(mocker, algo_name, target_device):
         },
         "target_device": target_device
     })
+    if COMPRESSION_ALGORITHMS.get(algo_name) == KnowledgeDistillationBuilder:
```

Comment:
Just do the comparison with the name of the algo as it is present in the config, no need to involve class objects.

Reply:
The thing is I can't get the name of KnowledgeDistillation algo from `KnowledgeDistillationBuilder` (except for `KnowledgeDistillationBuilder._registered_name`) so comparation between class objects seems appropriate.
`if algo_name == NoCompressionAlgorithmBuilder.__name__` works because `NoCompressionAlgorithmBuilder.__name__ == NoCompressionAlgorithmBuilder._registered_name` which is not the case for `KnowledgeDistillationBuilder`.

Or is it ok to `if algo_name == 'knowledge_distillation'`?

Reply:
Indeed, should be enough to do `if algo_name == 'knowledge_distillation'`

Reply:
FYI, after merging my PR `name` property will be available: 
https://github.com/openvinotoolkit/nncf/pull/682/files#diff-e8f054f8808cc9acc32c920dcec746b3f541b8448922f853419fdbe1d2462debR156-R158

we shouldn't forget to adapt it to the newer code base.